### PR TITLE
[RHOAIENG-2404] Replace remaining PF deprecated components

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Closes [RHOAIENG-2404](https://issues.redhat.com/browse/RHOAIENG-2404)

## Description
Remove use of Patternfly deprecated components

## How Has This Been Tested?
Tested manually

## Test Impact
The following components are effected:

- Data Science Pipelines -> Filter type select
- Experiments -> Experiments and runs -> click an experiment -> select a run -> Compare runs -> change filter to Experiment -> Search input
- Data Science Projects -> select a project -> workbench tab -> create a workbench -> Use a data connection -> use existing data connection -> Select data connection
- Data Science Projects -> select a project -> workbench tab -> create a workbench -> Cluster storage -> use existing persistent storage -> Select a persistent storage
- Science Projects -> select a project -> Cluster storage tab -> Add cluster storage -> connected workbench select

## Request review criteria:
Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

### CODE REVIEW

File: frontend/src/__tests__/cypress/cypress/pages/userManagement.ts [hunk 1]
```diff
@@ -38,12 +38,16 @@ class GroupSettingSection extends Contextual<HTMLElement> {
...
   }
 
-  findMultiGroupSelectButton() {
-    return this.find().findByRole('button', { name: 'Options menu' });
+  findMultiGroupSelectButton(admin: boolean) {
+    return this.find().findByRole('button', {
+      name: admin
+        ? 'Select the OpenShift groups that contain all Data Science administrators.'
+        : 'Select the OpenShift groups that contain all Data Science users.',
+    });
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/userManagement.ts:46]: This is odd.
It's the only button in the section and should use a common selector like a data-testid.
The scope of the object is already defined as admin or not, we shouldn't need to pass in an additional property to distinguish as such.

File: frontend/src/components/MultiSelection.tsx [hunk 2]
```diff
@@ -113,22 +182,25 @@ export const MultiSelection: React.FC<MultiSelectionProps> = ({ value, setValue
...
     }
     textInputRef.current?.focus();
   };
 
-  const noSelectedItems = value.filter((option) => option.selected).length === 0;
+  const noSelectedItems = allOptions.filter((option) => option.selected).length === 0;
 
   const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
     <MenuToggle
+      id={toggleId}
```
Comment (christianvogt) [inline frontend/src/components/MultiSelection.tsx:197]: We should also assign or support a separate data-testid attribute here.

File: frontend/src/concepts/roleBinding/RoleBindingPermissionsNameInput.tsx [hunk 3]
```diff
@@ -48,33 +47,27 @@ const RoleBindingPermissionsNameInput: React.FC<RoleBindingPermissionsNameInputP
     );
   }
 
+  const selectOptions = typeAhead.map((option) => ({
+    value: isProjectSubject ? namespaceToProjectDisplayName(option, projects) : option,
+    content: isProjectSubject ? namespaceToProjectDisplayName(option, projects) : option,
+  }));
```
Comment (christianvogt) [inline frontend/src/concepts/roleBinding/RoleBindingPermissionsNameInput.tsx:53]: is content necessary? won't it just render the value as content? if not, compute only once.
```suggestion
  const selectOptions = typeAhead.map((option) => {
    const value = isProjectSubject ? namespaceToProjectDisplayName(option, projects) : option;
    return {
      value,
      content: value,
    };
  });
```

File: frontend/src/pages/modelServing/screens/metrics/bias/BiasMetricConfigSelector.tsx [hunk 4]
```diff
@@ -26,71 +14,70 @@ const BiasMetricConfigSelector: React.FC<BiasMetricConfigSelectorProps> = ({
...
-            if (selected.find(byId(item))) {
-              // User has de-selected an item.
-              changeState(selected.filter(byNotId(item)));
-            } else {
-              // User has selected an item.
-              changeState([...selected, item]);
+      <div>ALL: {biasMetricConfigs.map((c) => c.name).join(', ')}</div>
+      <div>UI: {uiSelections}</div>
+      <div>current: {currentSelections}</div>
+      <div>initialSelections: {initialSelections.map((s) => s.name).join(', ')}</div>
```
Comment (andrewballantyne) [inline frontend/src/pages/modelServing/screens/metrics/bias/BiasMetricConfigSelector.tsx:32]: Debug code?

Comment (jeff-phillips-18) [reply frontend/src/pages/modelServing/screens/metrics/bias/BiasMetricConfigSelector.tsx:32]: Yup, not sure how i missed removing this ☹️

Comment (jeff-phillips-18) [reply frontend/src/pages/modelServing/screens/metrics/bias/BiasMetricConfigSelector.tsx:32]: Removing in https://github.com/opendatahub-io/odh-dashboard/pull/3376

**Pull request merged**